core/src/transport/memory.rs

```diff
@@ -68,7 +99,7 @@ impl Transport for MemoryTransport {
 
         let actual_addr = Protocol::Memory(port.get()).into();
 
-        let (tx, rx) = mpsc::unbounded();
+        let (tx, rx) = mpsc::channel(CHAN_BUF_SIZE);
```

Comment:
I think `2` is a good value. It's the number of maximum dialing connections that haven't been processed by the listener yet.
```suggestion
        let (tx, rx) = mpsc::channel(2);
```